hydromt/data_adapter/dataframe.py

```diff
@@ -176,54 +178,88 @@ def to_file(
-            fn_out = join(data_root, f"{data_name}.parquet")
-            obj.to_parquet(fn_out, **kwargs)
-        elif driver == "excel":
-            fn_out = join(data_root, f"{data_name}.xlsx")
-            obj.to_excel(fn_out, **kwargs)
-        else:
-            raise ValueError(f"DataFrame: Driver {driver} is unknown.")
+        try:
+            kwargs.pop("bbox", None)
+            obj = self.get_data(
+                time_tuple=time_tuple, variables=variables, logger=logger
+            )
 
-        return fn_out, driver, read_kwargs
+            if len(obj) == 0:
+                raise NoDataException()
+
+            read_kwargs = dict()
+            if driver is None or driver == "csv":
+                # always write as CSV
+                driver = "csv"
+                fn_out = join(data_root, f"{data_name}.csv")
+                obj.to_csv(fn_out, **kwargs)
+                read_kwargs["index_col"] = 0
+            elif driver == "parquet":
+                fn_out = join(data_root, f"{data_name}.parquet")
+                obj.to_parquet(fn_out, **kwargs)
+            elif driver == "excel":
+                fn_out = join(data_root, f"{data_name}.xlsx")
+                obj.to_excel(fn_out, **kwargs)
+            else:
+                raise ValueError(f"DataFrame: Driver {driver} is unknown.")
+
+            return fn_out, driver, read_kwargs
+        except NoDataException:
+            _exec_nodata_strat(
+                "No data to export", strategy=handle_nodata, logger=logger
+            )
+            return None
 
     def get_data(
         self,
-        variables=None,
-        time_tuple=None,
-        logger=logger,
-    ):
+        variables: Optional[Variables] = None,
+        time_tuple: Optional[TimeRange] = None,
+        handle_nodata: NoDataStrategy = NoDataStrategy.RAISE,
+        logger: Logger = logger,
+    ) -> Optional[pd.DataFrame]:
         """Return a DataFrame.
 
         Returned data is optionally sliced by time and variables,
         based on the properties of this DataFrameAdapter. For a detailed
         description see: :py:func:`~hydromt.data_catalog.DataCatalog.get_dataframe`
         """
-        # load data
-        fns = self._resolve_paths(variables)
-        df = self._read_data(fns, logger=logger)
-        self.mark_as_used()  # mark used
-        # rename variables and parse nodata
-        df = self._rename_vars(df)
-        df = self._set_nodata(df)
-        # slice data
-        df = DataFrameAdapter._slice_data(df, variables, time_tuple, logger=logger)
-        # uniformize data
-        df = self._apply_unit_conversion(df, logger=logger)
-        df = self._set_metadata(df)
-        return df
+        try:
+            # load data
+            fns = self._resolve_paths(variables=variables)
+            df = self._read_data(fns, handle_nodata=NoDataStrategy.RAISE, logger=logger)
+            if has_no_data(df):
+                raise NoDataException()
```

Comment:
Would it work if here you do ``_exec_nodata_strat`` instead or raise NoDataException? Just to avoid too many try / except

Reply:
so I prefer the try except because it means we do the handeling of this in only one place, so that makes it less likely to cause inconsisten behaviour later due to e.g. accidentally changing parameters. Do you mind the try/excepts? (I've reduced them a lot since our discussion)